content/en/docs/ops/ambient/upgrade/helm-upgrade/index.md

```diff
@@ -0,0 +1,129 @@
+
+    {{< text syntax=bash snip_id=update_helm >}}
+    $ helm repo update istio
+    {{< /text >}}
+
+## In-place upgrade
+
+You can perform an in place upgrade of Istio in your cluster using the Helm
+upgrade workflow.
+
+Before upgrading Istio, it is recommended to run the `istioctl x precheck` command to make sure the upgrade is compatible with your environment.
+
+{{< text syntax=bash snip_id=istioctl_precheck >}}
+$ istioctl x precheck
+✔ No issues found when checking the cluster. Istio is safe to install or upgrade!
+  To get started, check out <https://istio.io/latest/docs/setup/getting-started/>
+{{< /text >}}
+
+{{< warning >}}
+[Helm does not upgrade or delete CRDs](https://helm.sh/docs/chart_best_practices/custom_resource_definitions/#some-caveats-and-explanations) when performing an upgrade. Because of this restriction, an additional step is required when upgrading Istio with Helm.
+{{< /warning >}}
+
+### Manually upgrading the CRDs and Istio base chart
+
+1. Upgrade the Kubernetes custom resource definitions ({{< gloss >}}CRDs{{</ gloss >}}):
+
+    {{< text syntax=bash snip_id=manual_crd_upgrade >}}
+    $ kubectl apply -f manifests/charts/base/crds
+    {{< /text >}}
+
+1. Upgrade the Istio base chart:
+
+    {{< text syntax=bash snip_id=upgrade_base >}}
+    $ helm upgrade istio-base manifests/charts/base -n istio-system --skip-crds
+    {{< /text >}}
+
+### Upgrading the Istio discovery Component
+
+Istiod is the control plane component that manages and configures the proxies to route traffic within the ambient mesh.
+
+{{< text syntax=bash snip_id=upgrade_istiod >}}
+$ helm upgrade istiod istio/istiod -n istio-system
+{{< /text >}}
+
+### Upgrading the ztunnel component
+
+The ztunnel DaemonSet is the L4 node-proxy component of ambient.
+
+{{< warning >}}
+Upgrading ztunnel in-place will briefly disrupt all ambient mesh traffic on the node.
+Node cordoning and blue/green node pools are recommended to mitigate blast radius risk during production upgrades. See your Kubernetes provider documentation for details.
+{{< /warning >}}
+
+{{< text syntax=bash snip_id=upgrade_ztunnel >}}
+$ helm upgrade ztunnel istio/ztunnel -n istio-system
+{{< /text >}}
+
+### Upgrading the CNI Component
+
+The Istio CNI agent is responsible for detecting the pods that belong to the ambient mesh, and configuring the traffic redirection between pods and the ztunnel DaemonSet. It is not part of the data plane or control plane.
+
+The Istio CNI agent at version 1.x is compatible with control plane at version 1.x-1, 1.x, and 1.x+1, which means the Istio CNI agent and the Istio control plane can be upgraded independently and in any order, as long as their version difference is within one minor version.
+
+{{< warning >}}
+Upgrading the Istio CNI agent will reconfigure networking on the node, and as such will momentarily disrupt node traffic. To manage blast radius to application pods during Istio CNI agent upgrade, node cordons are recommended.
```

Comment:
Correct if you want to apply node cordons, you'll need at least 1 nodes, and possibly add extra nodes during upgrade.

Reply:
Yep. This is all highly dependent on your K8S setup, and it's a bit crude, but it's what we've settled on for the time being.

One of the goals in writing it down in black and white was to clarify what the actual current process for users is, so it's easier for people to reason about future improvements.

Reply:
can you point out the detail in the doc? At latest we should highlight it.

Reply:
I mention it at the top of the page here: https://github.com/istio/istio.io/pull/14491/files#diff-1e0e52b46d7e94ba4f52d12e3a86d71d14b68b0ade7c78618ef54d81f23d6d18R19

if that's what you're asking.

As I mentioned in the PR description, I don't want to get into the weeds of how this works operationally in this doc, because it is different for every cloud provider, and not easily demo-able locally with `kind` or similar.

That is an inherent downside of this approach, it's implicitly a punt and very cloud-provider dependent.

Reply:
ok, it's ok for now.